### PR TITLE
Merge NullableAnnotation.NotApplicable and Disabled to None, and rename NullableFlowState.NotApplicable to None for consistency

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/37441. @dotnet/roslyn-compiler for review. /cc @jasonmalinowski @ryzngard

### CODE REVIEW

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -55,10 +55,9 @@ Microsoft.CodeAnalysis.NullabilityInfo.Annotation.get -> Microsoft.CodeAnalysis.
 Microsoft.CodeAnalysis.NullabilityInfo.Equals(Microsoft.CodeAnalysis.NullabilityInfo other) -> bool
 Microsoft.CodeAnalysis.NullabilityInfo.FlowState.get -> Microsoft.CodeAnalysis.NullableFlowState
 Microsoft.CodeAnalysis.NullableAnnotation
-Microsoft.CodeAnalysis.NullableAnnotation.Annotated = 3 -> Microsoft.CodeAnalysis.NullableAnnotation
-Microsoft.CodeAnalysis.NullableAnnotation.Disabled = 1 -> Microsoft.CodeAnalysis.NullableAnnotation
```
Comment (gafter) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: >Disabled [](start = 42, length = 8)

Isn't this API already released publicly?

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -76,7 +75,7 @@ Microsoft.CodeAnalysis.NullableContextOptions.Warnings = 1 -> Microsoft.CodeAnal
 Microsoft.CodeAnalysis.NullableContextOptionsExtensions
 Microsoft.CodeAnalysis.NullableFlowState
 Microsoft.CodeAnalysis.NullableFlowState.MaybeNull = 2 -> Microsoft.CodeAnalysis.NullableFlowState
-Microsoft.CodeAnalysis.NullableFlowState.NotApplicable = 0 -> Microsoft.CodeAnalysis.NullableFlowState
```
Comment (gafter) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: >NotApplicable [](start = 41, length = 13)

Same for this

**Comment by gafter** [comment]
> *REMOVED*Microsoft.CodeAnalysis.Operations.IEventAssignmentOperation.EventReference.get -> Microsoft.CodeAnalysis.Operations.IEventReferenceOperation

All the stull that is published should be moved to the PublicAPI.Shipped.txt file.

---
Refers to: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt:1 in 0dab49c. [](commit_id = 0dab49c75e5c84d1f4c1e279db0d18b814eb00be, deletion_comment = False)

**Comment by gafter** [comment]
There was a review comment not being displayed by CodeFlow:

These look like incompatible changes to shipped APIs. If that is what is intended it needs to happen on a 16.3 branch.

**Comment by gafter** [comment]
I will chat with Andy tomorrow to make sure I understand our branch and bar status.

---
In reply to: [514440025](https://github.com/dotnet/roslyn/pull/37442#issuecomment-514440025) [](ancestors = 514440025)

**Comment by cston** [comment]
>             TopLevelDisabled = TopLevelAnnotated | TopLevelNotAnnotated,

None?

---
Refers to: src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs:32 in 0dab49c. [](commit_id = 0dab49c75e5c84d1f4c1e279db0d18b814eb00be, deletion_comment = False)

File: src/Compilers/Core/Portable/Symbols/NullableAnnotation.cs [hunk 3]
```diff
@@ -13,13 +13,7 @@ public enum NullableAnnotation : byte
         /// The expression has not been analyzed, or the syntax is
         /// not an expression (such as a statement).
         /// </summary>
-        NotApplicable = 0,
-        /// <summary>
-        /// The expression comes from a library not updated to C# 8
-        /// or nullable was disabled for the definition, and has no
```
Comment (cston) [inline src/Compilers/Core/Portable/Symbols/NullableAnnotation.cs]: This case still applies.

File: src/Compilers/VisualBasic/Test/Semantic/Compilation/CompilationAPITests.vb [hunk 4]
```diff
@@ -1523,12 +1523,12 @@ BC2014: the value '_' is invalid for option 'RootNamespace'
...
-            Assert.Equal(CodeAnalysis.NullableAnnotation.NotApplicable, comp.CreateArrayTypeSymbol(elementType, elementNullableAnnotation:=CodeAnalysis.NullableAnnotation.NotAnnotated).ElementNullableAnnotation)
-            Assert.Equal(CodeAnalysis.NullableAnnotation.NotApplicable, comp.CreateArrayTypeSymbol(elementType, elementNullableAnnotation:=CodeAnalysis.NullableAnnotation.Annotated).ElementNullableAnnotation)
+            Assert.Equal(CodeAnalysis.NullableAnnotation.None, comp.CreateArrayTypeSymbol(elementType).ElementNullableAnnotation)
+            Assert.Equal(CodeAnalysis.NullableAnnotation.None, comp.CreateArrayTypeSymbol(elementType, elementNullableAnnotation:=Nothing).ElementNullableAnnotation)
+            Assert.Equal(CodeAnalysis.NullableAnnotation.None, comp.CreateArrayTypeSymbol(elementType, elementNullableAnnotation:=CodeAnalysis.NullableAnnotation.None).ElementNullableAnnotation)
+            Assert.Equal(CodeAnalysis.NullableAnnotation.None, comp.CreateArrayTypeSymbol(elementType, elementNullableAnnotation:=CodeAnalysis.NullableAnnotation.None).ElementNullableAnnotation)
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Semantic/Compilation/CompilationAPITests.vb]: >Assert.Equal(CodeAnalysis.NullableAnnotation.None, comp.CreateArrayTypeSymbol(elementType, elementNullableAnnotation:=CodeAnalysis.NullableAnnotation.None).ElementNullableAnnotation) [](start = 12, length = 182)

This is a now a duplicate of the line above.

**Comment by 333fred** [comment]
`master` is still 16.3.

---
In reply to: [514445903](https://github.com/dotnet/roslyn/pull/37442#issuecomment-514445903) [](ancestors = 514445903,514440025)

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt (hunk 1 above)
Comment (333fred) [reply src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: It's released as a preview API, which we have previously said was subject to change. This is the last chance we have to make breaking changes here.

---
In reply to: [306591815](https://github.com/dotnet/roslyn/pull/37442#discussion_r306591815) [](ancestors = 306591815)

File: src/Compilers/Core/Portable/Symbols/NullableAnnotation.cs (hunk 3 above)
Comment (333fred) [reply src/Compilers/Core/Portable/Symbols/NullableAnnotation.cs]: I felt like the original comment on NotApplicable was actually sufficient (namely "the expression has not been analyzed"), but I can add these as well if you'd prefer.

---
In reply to: [306882877](https://github.com/dotnet/roslyn/pull/37442#discussion_r306882877) [](ancestors = 306882877)

**Comment by jcouv** [comment]
@333fred Conflicts here too

**Comment by jcouv** [comment]
@333fred Go ahead and merge once CI is green.

**Pull request merged**